src/pal/src/exception/seh-unwind.cpp

```diff
@@ -73,17 +73,17 @@ struct __cxa_exception
 // Virtual Unwinding
 //----------------------------------------------------------------------
 
-static void WinContextToUnwindContext(CONTEXT *winContext, unw_context_t *unwContext)
+static void WinContextToUnwindContext(CONTEXT *winContext, unw_cursor_t *cursor)
 {
 #if defined(_AMD64_)
-    unwContext->uc_mcontext.gregs[REG_RIP] = winContext->Rip;
-    unwContext->uc_mcontext.gregs[REG_RSP] = winContext->Rsp;
-    unwContext->uc_mcontext.gregs[REG_RBP] = winContext->Rbp;
-    unwContext->uc_mcontext.gregs[REG_RBX] = winContext->Rbx;
-    unwContext->uc_mcontext.gregs[REG_R12] = winContext->R12;
-    unwContext->uc_mcontext.gregs[REG_R13] = winContext->R13;
-    unwContext->uc_mcontext.gregs[REG_R14] = winContext->R14;
-    unwContext->uc_mcontext.gregs[REG_R15] = winContext->R15;
+    unw_set_reg(cursor, UNW_REG_IP, winContext->Rip);
+    unw_set_reg(cursor, UNW_REG_SP, winContext->Rsp);
```

Comment:
How could setting IP update SP? There can be possibly many different frames that have the same IP - consider recursive functions.